### PR TITLE
Run `pip-compile` for Each Extra

### LINKED ISSUE #21: Create `requirements-*.txt` for each `options.extras_require`
We need to also document what packages (versions) are used for the different flavors of a package.

### PR DESCRIPTION
**Pull request by ric-evans**
closes #21 

Also, respect `directory-exclude` regardless if `package_dirs` is given

### CODE REVIEW

**Comment by ric-evans** [comment]
You can see it running working here: https://github.com/WIPACrepo/MQClient/pull/50

File: list_extras.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+if __name__ == "__main__":
+
+    def _type_setup_cfg(arg: str) -> Path:
+        fpath = Path(arg)
+        if fpath.name != "setup.cfg":
+            raise ValueError()  # excepted by argparse & formatted nicely
+        if not fpath.exists():
+            raise FileNotFoundError(arg)
```
Comment (dsschult) [inline list_extras.py:27]: Note that `FileNotFoundError` won't be processed by [argparse](https://docs.python.org/3/library/argparse.html#type) correctly for a help message.
> The argument to type can be any callable that accepts a single string. If the function raises ArgumentTypeError, [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), or [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError), the exception is caught and a nicely formatted error message is displayed. No other exception types are handled.

**Review by dsschult — COMMENTED** [review]
Generally looks good.

File: list_extras.py (hunk 1 above)
Comment (ric-evans) [reply list_extras.py:27]: Wait, does this mean any other Exception is caught and ignored, or just raised to the user verbatim (just not in "a nicely formatted error message")?

Comment (dsschult) [reply list_extras.py:27]: I believe the latter.  It won't print the help message, just raise a std exception trace.

Comment (ric-evans) [reply list_extras.py:27]: Gotcha. I'm going to use wipac dev tools, after this issue: https://github.com/WIPACrepo/wipac-dev-tools/pull/43

**Pull request merged**